portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,78 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.lang.Integer;
+import java.util.ArrayList;
+import java.util.Hashtable;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content. TODO: modify this file to handle comments data. I have to redo everything from step 5*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Integer num_req = Integer.parseInt(request.getParameter("quantity"));
+
+        Query query = new Query("Comment");
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        ArrayList<String> storedmessages = new ArrayList<String>();
+        for (Entity entity : results.asIterable()) {
+            String userComment = (String) entity.getProperty("comment");
+            
+            if(storedmessages.size() < num_req){
```

Comment:
Instead of this if/ else break here, you could pass in FetchOptions with a limit of num_req) set into the asIterable call (https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery.html#asIterable-com.google.appengine.api.datastore.FetchOptions-)

Some examples are here:
https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/FetchOptions.html

Reply:
I added that function in my code. I can just commit it for the next PR, right? Also this is a really great tool thank you for showing me this!

Reply:
I think if you add more commits to this branch it will show up on this PR.

Reply:
Ok great will do!

Reply:
I can now see the commit I made to this section it's my 10th commit. I think since this comment thread is labeled as outdated, you should be able to see the change I made now as well.

Reply:
Cool! I see the changes now. Just a couple more small comments to address (The DeleteServelet and one on index.html) and I'll approve :)